docker/opbeans/rum/tasks.js

```diff
@@ -10,30 +10,34 @@ function sleep(ms) {
 }
 
 async function run() {
-    const chromeless = new Chromeless({
-        lauchChrome: true
-    })
+    try {
+        const chromeless = new Chromeless({
+            lauchChrome: true
+        })
 
-    for (; ;) {
-        url = await chromeless
-            .goto(url)
-            .evaluate((baseUrl) => {
-                // activateLoadGeneration is defined in opbeans-frontend
-                if (typeof window.activateLoadGeneration === 'function') {
-                    console.log('Activating route change load generation')
-                    window.activateLoadGeneration()
-                }
+        for (; ;) {
+            url = await chromeless
+                .goto(url)
+                .evaluate((baseUrl) => {
+                    // activateLoadGeneration is defined in opbeans-frontend
+                    if (typeof window.activateLoadGeneration === 'function') {
+                        console.log('Activating route change load generation')
+                        window.activateLoadGeneration()
+                    }
 
-                var links = document.querySelectorAll('a[href^="/"]')
-                if (links && links.length) {
-                    var i = Math.floor(Math.random() * links.length)
-                    return links[i].href
-                } else {
-                    return baseUrl
-                }
-            }, baseUrl);
-        console.log(url)
-        await sleep(8000 + Math.floor(Math.random() * 10000))
+                    var links = document.querySelectorAll('a[href^="/"]')
+                    if (links && links.length) {
+                        var i = Math.floor(Math.random() * links.length)
+                        return links[i].href
+                    } else {
+                        return baseUrl
+                    }
+                }, baseUrl);
+            console.log(url)
+            await sleep(8000 + Math.floor(Math.random() * 10000))
+        }
+    } catch (e){
+      run();
```

Comment:
This is a recursion, right? If so, I'd recommend having a base case with an exit condition to avoid infinite loops.

I know that the pipeline will kill it, but just to make the code resilient and decoupled from the build tool